conduit/pipeline.go

```diff
@@ -228,10 +226,27 @@ func (p *pipelineImpl) Start() error {
 		p.logger.Infof("Initialized Processor: %s", processorName)
 	}
 
-	return p.RunPipeline()
+	go p.runPipeline()
+	return err
```

Comment:
I missed this in my first pass. I think it's better to let the caller decide when to start a go routine, that way it's more natural to control a graceful shutdown with sync.WaitGroup.

It is nice that the pidFile/profFile tests are deterministic this way.

Maybe it would be the best of both solutions to split Init/Start by renaming:
* `Start` -> `Initialize`
* `runPipeline` -> `Start`

Reply:
Moved all of the goroutine specific code into `Start` and added a synchronous `Init` function. Also added a blocking `Wait` function to the pipeline so that we don't have an empty loop places where we have no other blocking operations to wait on.

Slightly modified the error variable used in the handlers, but like you've said I don't think there's an idiomatic way to achieve that other than just protecting reads/writes with a mutex. 

Let me know what you think...I tried your suggestion above of moving wait group management into the conduit daemon, but it ended up splitting up the wait group and context handling/blocking in ways that I didn't like.